lib/pmdtester/builders/diff_builder.rb

```diff
@@ -141,7 +156,7 @@ def initialize(violation, branch)
     def eql?(other)
       @violation['beginline'].eql?(other.violation['beginline']) &&
```

Comment:
does this work or shouldn't that be rather "@violation.at_xpath('beginline')"? Same then for the 'rule' attribute.

Reply:
Using `at_xpath` in `@error.at_xpath('filename').text` because `filename` is a subelement of `error` element.
`beginline` and `rule` is a attribute of `violation` element, so we should use `[attributename]` to get the value of attribute.